test/nodes/test_retriever.py

```diff
@@ -81,6 +82,46 @@ def test_retrieval(retriever_with_docs: BaseRetriever, document_store_with_docs:
         assert len(result) == 0
 
 
+class MockBaseRetriever(MockRetriever, BaseRetriever):
```

Comment:
The very last thing: MockRetriever already inherits from BaseRetriever:

```suggestion
class MockBaseRetriever(MockRetriever):
```